build.gradle

```diff
@@ -24,39 +19,36 @@ allprojects {
 }
 
 ext {
-  minSdkVersion = 14
+  minSdkVersion = 21
   targetSdkVersion = 27
-  compileSdkVersion = 27
+  compileSdkVersion = 29
   sourceCompatibilityVersion = JavaVersion.VERSION_1_8
   targetCompatibilityVersion = JavaVersion.VERSION_1_8
 }
 
-def supportLibVersion = '27.1.1'
-
 ext.deps = [
   // Android
-  android               : 'com.google.android:android:4.1.1.4',
-  support_v4            : 'com.google.android:support-v4:r7',
-  preference_v7         : "com.android.support:preference-v7:${supportLibVersion}",
-  appcompat_v7          : "com.android.support:appcompat-v7:${supportLibVersion}",
-  support_annotations   : "com.android.support:support-annotations:${supportLibVersion}",
+  android               : "com.google.android:android:4.1.1.4",
+  preference_x          : "androidx.preference:preference:1.1.0",
+  appcompat_x           : "androidx.appcompat:appcompat:1.1.0",
+  annotations_x         : "androidx.annotation:annotation:1.1.0",
   // Processor
   javapoet              : 'com.squareup:javapoet:1.6.1',
 
   // Test dependencies
   junit                 : 'junit:junit:4.12',
   mockito               : 'org.mockito:mockito-all:1.10.19',
   compiletesting        : 'com.google.testing.compile:compile-testing:0.7',
-  roboelectric          : 'org.robolectric:robolectric:3.8',
-  robolectric_support   : 'org.robolectric:shadows-supportv4:3.8',
-  equalsverifier        : 'nl.jqno.equalsverifier:equalsverifier:2.1.8'
+  roboelectric          : 'org.robolectric:robolectric:4.3.1',
```

Comment:
version 4.3.x has issues, most notably: https://github.com/robolectric/robolectric/issues/5245
I wonder if it's a problem also in this project...

Reply:
I didn't notice any issue with this version, maybe it depends a more complex setup like the one we have in the listener app